### PR TITLE
Add Palm Testnet Deployment Scripts

### PR DESCRIPTION
**Pull request by lindsgil**
Added deployment scripts for Palm Testnet, updated scripts in posterity directory to work with current file structure and tested with contracts deployed to the Palm Testnet network

### CODE REVIEW

File: scripts/PBAB+Collabs/palm-demo/1_engine_demo_contract_suite_deployer_palm_testnet.ts [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+  // Output instructions for manual Etherscan verification.
+  const standardVerify = "yarn hardhat verify";
+  console.log(`Verify core contract deployment with:`);
+  console.log(
+    `${standardVerify} --network ${networkName} ${genArt721CoreFlex.address} "${tokenName}" "${tokenTicker}" ${randomizerAddress}`
+  );
+  console.log(`Verify Minter deployment with:`);
+  console.log(
+    `${standardVerify} --network ${networkName} ${genArt721Minter.address} ${genArt721CoreFlex.address}`
+  );
```
Comment (ryley-o) [inline scripts/PBAB+Collabs/palm-demo/1_engine_demo_contract_suite_deployer_palm_testnet.ts:157]: does verification via etherscan work? or is there a different verification process that we need to go through? If different, I would vote to remove this section from this script.

**Review by ryley-o — APPROVED** [review]
Thank you @lindsgil! 🙏 

I don't think we _need_ to update the posterity scripts here, but seems like a fine idea to update them 👍 

If we deployed some contracts that we care about keeping track of, I would recommend we keep a `DEPLOYMENTS.md` somewhere. I think this was all proof-of-concept though, right? so maybe nothing needs to be noted.

File: scripts/PBAB+Collabs/palm-demo/1_engine_demo_contract_suite_deployer_palm_testnet.ts [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+  // Allowlist AB staff
+  console.log(`Detected testnet - Adding AB staff to the whitelist.`);
+  const devAddresses = [
+    "0xB8559AF91377e5BaB052A4E9a5088cB65a9a4d63", // purplehat
+    "0x3c3cAb03C83E48e2E773ef5FC86F52aD2B15a5b0", // dogbot
+    "0x0B7917b62BC98967e06e80EFBa9aBcAcCF3d4928", // ben_thank_you
+    "0x3b9038fa89783CBA1933c1689043b4dae2032d1c", // heylinds
+    "0x48631d49245185cB38C01a31E0AFFfA75c133d9a", // meck
+  ];
```
[the remark this answers is not in the document]
Comment (lyaunzbe) [reply scripts/PBAB+Collabs/palm-demo/1_engine_demo_contract_suite_deployer_palm_testnet.ts:110]: perhaps via `providers.getNetwork` and checking the chainId?

File: scripts/PBAB+Collabs/palm-demo/1_engine_demo_contract_suite_deployer_palm_testnet.ts (hunk 1 above)
Comment (lyaunzbe) [reply scripts/PBAB+Collabs/palm-demo/1_engine_demo_contract_suite_deployer_palm_testnet.ts:157]: i dont think palm has an etherscan based block explorer, only https://explorer.palm.io/ as far as im aware. i noticed they support something called sourcify, perhaps something that can still be automated (at a later time): https://docs.sourcify.dev/docs/intro

**Comment by lyaunzbe** [comment]
@ryley-o agree overall re having that `DEPLOYMENTS` doc, though as you mentioned for this one it's more of an internal demo vs an actual partner deployment we are tracking so I think we can forego, but also no strong opinion on that from my side

File: scripts/PBAB+Collabs/palm-demo/1_engine_demo_contract_suite_deployer_palm_testnet.ts (hunk 1 above)
Comment (ryley-o) [reply scripts/PBAB+Collabs/palm-demo/1_engine_demo_contract_suite_deployer_palm_testnet.ts:157]: It looks like this community hardhat plugin might do the trick! https://www.npmjs.com/package/hardhat-sourcify
@lindsgil - do you mind seeing if that plugin meets our needs here?

**Review by mchrupcala — APPROVED** [review]
Looks awesome!

File: hardhat.config.ts [hunk 3]
```diff
@@ -8,12 +8,18 @@ import "hardhat-contract-sizer";
...
 
+
 const MAINNET_JSON_RPC_PROVIDER_URL = process.env.MAINNET_JSON_RPC_PROVIDER_URL;
 const GOERLI_JSON_RPC_PROVIDER_URL = process.env.GOERLI_JSON_RPC_PROVIDER_URL;
 const MAINNET_PRIVATE_KEY = process.env.MAINNET_PRIVATE_KEY;
 const TESTNET_PRIVATE_KEY = process.env.TESTNET_PRIVATE_KEY;
 const ETHERSCAN_API_KEY = process.env.ETHERSCAN_API_KEY;
 
+
+// L2 Configuration
```
[the remark this answers is not in the document]
Comment (jakerockland) [reply hardhat.config.ts:19]: +++

**Comment by lindsgil** [comment]
> Awesome!! 🔥 🙏
> 
> One ask: could we make these deployer scripts reference deploying the base Engine template by default rather than the Flex one?

Good call! Updated to use the base Engine template 🙏

**Comment by lindsgil** [comment]
> Thank you @lindsgil! 🙏
> 
> I don't think we _need_ to update the posterity scripts here, but seems like a fine idea to update them 👍
> 
> If we deployed some contracts that we care about keeping track of, I would recommend we keep a `DEPLOYMENTS.md` somewhere. I think this was all proof-of-concept though, right? so maybe nothing needs to be noted.

Thanks @ryley-o !! I added a `DEPLOYMENTS.md` file with block explorer links to the test minter and core contracts with a note that these were test uploads

File: scripts/PBAB+Collabs/palm-demo/1_engine_demo_contract_suite_deployer_palm_testnet.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (lindsgil) [reply scripts/PBAB+Collabs/palm-demo/1_engine_demo_contract_suite_deployer_palm_testnet.ts]: Good call! Added a check for the chainId

File: hardhat.config.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (lindsgil) [reply hardhat.config.ts]: Awesome! Thank you for calling this out! Updated :)

File: scripts/PBAB+Collabs/palm-demo/1_engine_demo_contract_suite_deployer_palm_testnet.ts (hunk 1 above)
Comment (lindsgil) [reply scripts/PBAB+Collabs/palm-demo/1_engine_demo_contract_suite_deployer_palm_testnet.ts]: Thanks for sending that plugin @ryley-o ! I'm getting some weird behavior when I install it.. I'll sidebar you and Ben on slack and see if y'all have any ideas. I was able to "partially" verify the contract through the Sourcify.dev UI which opened up the "Read Contract" and "Write Contract" tabs in the block explorer (check it out here: https://explorer.palm-uat.xyz/address/0x4C7E1e1Ba5e35934C8323d75AF838fDa26588177/contracts#address-tabs) .. Did this by zipping the build-info json file with the sol contract file and submitting here: https://sourcify.dev/#/verifier

**Pull request merged**